### ISSUE TITLE
not all entries being set in a list

### ISSUE THREAD

**Issue by ham1255** [SkriptLang/Skript#5570]
### Skript/Server Version

```markdown
> skript info
[19:22:33 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[19:22:33 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[19:22:33 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[19:22:33 INFO]: [Skript] Server Version: git-Purpur-1949 (MC: 1.19.4)
[19:22:33 INFO]: [Skript] Skript Version: 2.6.4
[19:22:33 INFO]: [Skript] Installed Skript Addons: None
[19:22:33 INFO]: [Skript] Installed dependencies: None
```

### Bug Description

the code provided in `steps to reproduce` should return "1652 VS 1652", but it always returns a smaller number that is different if spamming the function, for example "1636 VS 1652". %random uuid% should be unique enough to not run into multiple collisions every time, and based on my testing this is not UUID collision, but something else, it just ends up being a different amount of entries in the list due to unknown reason?

### Expected Behavior

should return `1652 VS 1652` instead of `*randomnumber* VS 1652`

### Steps to Reproduce

load this code:
```
function uuidconflict():
	loop 1652 times:
		add 1 to {_uuidlooped}
		set {_insert::%random uuid%} to "test"
	send "%amount of {_insert::*}% VS %{_uuidlooped}%" to console

```
console output when using the function:
```
> !uuidconflict()
[19:28:06 INFO]: executing 'uuidconflict()'
[19:28:06 INFO]: 1640 VS 1652
> !uuidconflict()
[19:28:07 INFO]: executing 'uuidconflict()'
[19:28:07 INFO]: 1636 VS 1652
> !uuidconflict()
[19:28:07 INFO]: executing 'uuidconflict()'
[19:28:07 INFO]: 1639 VS 1652
> !uuidconflict()
[19:28:08 INFO]: executing 'uuidconflict()'
[19:28:08 INFO]: 1639 VS 1652
> !uuidconflict()
[19:28:08 INFO]: executing 'uuidconflict()'
[19:28:08 INFO]: 1635 VS 1652
> !uuidconflict()
[19:28:09 INFO]: executing 'uuidconflict()'
[19:28:09 INFO]: 1634 VS 1652
> !uuidconflict()
[19:28:09 INFO]: executing 'uuidconflict()'
[19:28:09 INFO]: 1637 VS 1652
> !uuidconflict()
[19:28:10 INFO]: executing 'uuidconflict()'
[19:28:10 INFO]: 1636 VS 1652
> !uuidconflict()
[19:28:10 INFO]: executing 'uuidconflict()'
[19:28:10 INFO]: 1637 VS 1652
> !uuidconflict()
[19:28:11 INFO]: executing 'uuidconflict()'
[19:28:11 INFO]: 1635 VS 1652

```

### Other

* it started in skript 2.6.4 specially in this pull request #5170 possibility `VariableMap` class
* using `add "" to {_list::*}` works 

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by DelayedGaming** [comment]
Unable to replicate bug.
Tested with Skript 2.7.0-beta2 on a Paper 1.19.4 server with a modified code for debugging.
Can you try my code and see if your issue persists, before and updating to Skript 2.7.0-beta2?
```vb
function a(max: int):
    loop {_max} times:
        set {_u} to random uuid
        if {_list::%{_u}%} is set:
            add 1 to {_dupe}
        set {_list::%{_u}%} to 1
    send "&a]&r Input %{_max}%, size %size of {_list::*}%, dupe %{_dupe} ? 0%" to console
```
![IMG_0776](https://user-images.githubusercontent.com/72163224/229071893-808d0ac9-fda7-48b8-b852-4642505f0d4b.jpeg)

**Comment by TheLimeGlass** [comment]
There are duplicate uuids being added, there is a small chance that the same UUID is generated still, so it'll override the existing, making your list count smaller. You should check that the newly random uuid is not already in the system. DelayedGaming didn't replicate the issue because they're checking for duplicates.

**Comment by ham1255** [comment]
I have checked if there duplicate but no there wasn't any duplicates

**Comment by DelayedGaming** [comment]
It is only close 2k iterations. It is a step to impossible to generate duplicate UUIDs in such, unless _you’re the chosen one for the next lottery_.

**Comment by TheLimeGlass** [comment]
Or their system random is messed with

**Comment by ham1255** [comment]
At first person who told me about this i said might be addon issues and so i made fresh server with skript 2.6.4

**Comment by TheLimeGlass** [comment]
use `size of` over `amount of` for variables

**Comment by DelayedGaming** [comment]
#### Successfully replicated bug:
Skript: **Positive on 2.6.4**, _negative on Skript 2.7.0-beta2_
Platform: Paper 1.19.4, Purpur 1.19.3
Personal investigation shows that the list size is indeed smaller than expected, but no null or duplicate UUIDs detected, but appears to be fixed in 2.7.0 beta. @ham1255 Please use Skript 2.7.0-beta2 if able.

#### Code used:
```vb
function a(max: int):
    loop {_max} times:
        delete {_u}
        set {_u} to random uuid
        if {_u} is not set:
            add 1 to {_null}
            send "&eUUID Variable is null" to console
        else if {_list::%{_u}%} is set:
            add 1 to {_dupe}
            send "&cDUPE FOUND %{_u}%" to console
        set {_list::%{_u}%} to 1
    send "&a]&r Input %{_max}%, size %size of {_list::*}%, dupe %{_dupe} ? 0%, random uuid expr null %{_null} ? 0% occurence" to console
```
![IMG_0781](https://user-images.githubusercontent.com/72163224/229099785-8cee49fe-ad22-4b77-9092-40c360779204.jpeg)

**Comment by Moderocky** [comment]
This issue is fixed in the latest beta release.
